src/main/java/core/basesyntax/service/impl/TransactionsListServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransactionsListService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class TransactionsListServiceImpl implements TransactionsListService {
+    private static final int OPERATION_TYPE_INDEX = 0;
+    private static final int FRUIT_TYPE_INDEX = 1;
+    private static final int AMOUNT_INDEX = 2;
+    private static final String COMMA_DELIMITER = ",";
+
+    @Override
+    public List<FruitTransaction> getTransactionsList(String lines, Map<String, Integer> fruitMap) {
+        List<FruitTransaction> fruitTransactionsList = new ArrayList<>();
+        String[] splitLines = lines.split(System.lineSeparator());
+        for (int i = 1; i < splitLines.length; i++) {
```

Comment:
if 'i' is not used for anything but index access you should just use for each
also you could simplify this method a lot by using streams

Reply:
fixed